### PR TITLE
Only cast logits to float when computing loss

### PR DESCRIPTION
**Pull request by ringohoffman**
# What does this PR do?

A few updates that were missed in #31292 and #33902.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

cc: @Cyrilvallez @gante @ArthurZucker

### CODE REVIEW

**Review by gante — APPROVED** [review]
LGTM, thank you for the fix 🙏

File: src/transformers/models/chameleon/modeling_chameleon.py [hunk 1]
```diff
@@ -1595,7 +1595,9 @@ def forward(
 
         hidden_states = outputs[0]
         logits = self.lm_head(hidden_states)
-        logits = logits.float()
+        if labels is not None:
+            # Upcast to float if we need to compute the loss to avoid potential precision issues
+            logits = logits.float()
```
[the remark this answers is not in the document]
Comment (ringohoffman) [reply src/transformers/models/chameleon/modeling_chameleon.py]: Okay, moved it for you.

**Review by ArthurZucker — APPROVED** [review]
🤗 thanks for bookeeping hehe
